tokio/src/sync/rwlock.rs

```diff
@@ -11,10 +11,10 @@ pub(crate) use write_guard::RwLockWriteGuard;
 pub(crate) use write_guard_mapped::RwLockMappedWriteGuard;
 
 #[cfg(not(loom))]
-const MAX_READS: usize = 32;
+const MAX_READS: u32 = std::u32::MAX >> 3;
```

Comment:
The other PR set this to `batch_semaphore::MAX_PERMITS`. How much larger/smaller is this?

Reply:
Well.. it is A LOT smaller than batch_semaphore::MAX_PERMITS. However, the idea is that we cannot acquire anything more than `u32` from the semaphore so I think this is the right number. The shit mimics the one of `batch_semaphore::MAX_PERMITS`  because on 32bit the last three bits are reserved for state. Does that sound correct?

Reply:
I mean, the number here is half a billion, so it's probably large enough. However, we were having some trouble in the other PR after increasing it. It would be nice to run this on a 32-bit machine to see if that also happens with this one.

Reply:
I think the problems with the other PR were due to this [test](https://github.com/tokio-rs/tokio/blob/a7c94aadb08d4ee41109207b6c9d5e1758234d25/tokio/tests/sync_rwlock.rs#L57) timing out. Will try and run it on 32bit to verify though.

Reply:
@Darksonn I did run it on 32bit and it all seems fine. I think at the end the problem was at a faulty assert in the semaphore code.